### PR TITLE
Excel merged cells test

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt [hunk 1]
```diff
@@ -26,6 +26,9 @@ import io.kotest.matchers.shouldBe
 
 import java.io.File
 
+import org.apache.poi.ss.usermodel.Cell
```
Comment (mnonnenmacher) [inline reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt:29]: I don't understand the commit message, what is the actual problem solved by this change? Is it just a preparation for the next commit? How is resaving the file related to this?

Comment (sschuberth) [reply reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt:29]: So, the change is complete unrelated to the next commit; it's just a preparation to be able to conveniently update `file-counter-expected-scan-report.xlsx` with Excel, if we ever need to do so.

Comment (mnonnenmacher) [reply reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt:29]: Thanks for the explanation.

File: reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt [hunk 2]
```diff
@@ -52,25 +53,35 @@ class ExcelReporterFunTest : WordSpec({
...
 
                 actualRowNums shouldContainExactly expectedRowNums
-                actualRowNums.map { "Row" to it }.forAll { (_, rowNum) ->
-                    val actualRow = actualSheet.getRow(rowNum)
-                    val actualColumnIndices = actualSheet.uniqueCellIterator(actualRow).map { it.columnIndex }.toList()
-
-                    val expectedRow = expectedSheet.getRow(rowNum)
-                    val expectedColumnIndices = actualSheet.uniqueCellIterator(expectedRow).map { it.columnIndex }.toList()
+                actualRowNums.forAll { rowAddress ->
+                    // The non-null assertion is fine as we pass "it" as a row directly above.
```
Comment (mnonnenmacher) [inline reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt:69]: Which "it" do you refer to?

File: reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt [hunk 3]
```diff
@@ -52,25 +53,35 @@ class ExcelReporterFunTest : WordSpec({
...
+                    val expectedColumnIndices = actualSheet.uniqueCellIterator(expectedRow).map {
+                        rowAddress.copy(column = it.columnIndex)
+                    }.toList()
 
                     actualColumnIndices shouldContainExactly expectedColumnIndices
-                    actualColumnIndices.map { Triple(sheetName, rowNum, it) }.forAll { (_, _, columnIndex) ->
-                        val actualCell = actualRow.getCell(columnIndex)
-                        val expectedCell = expectedRow.getCell(columnIndex)
+                    actualColumnIndices.forAll { address ->
+                        // The non-null assertion is fine as we pass "it" as a column directly above.
```
Comment (mnonnenmacher) [inline reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt:82]: Which "it" do you refer to?

File: reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt (hunk 2 above)
Comment (sschuberth) [reply reporter/src/funTest/kotlin/reporters/ExcelReporterFunTest.kt:69]: Ah, sorry, that is a left-over from a previous iteration. Will fix.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**